File: php/src/Services/UserDepositService.php
<?php

namespace CCPayment\Services;

use CCPayment\Client;
use CCPayment\APIError;
use GuzzleHttp\Exception\GuzzleException;

class UserDepositService
{
    private Client $client;

    public function __construct(Client $client)
    {
        $this->client = $client;
    }

    /**
     * @throws APIError
     * @throws GuzzleException
     */
    public function getOrCreateUserDepositAddress(array $data): array
    {
        return $this->client->post('/getOrCreateUserDepositAddress', $data);
    }

    /**
     * @throws APIError
     * @throws GuzzleException
     */
    public function getUserDepositRecord(string $recordId): array
    {
        return $this->client->post('/getUserDepositRecord', ['recordId' => $recordId]);
    }

    /**
     * @throws APIError
     * @throws GuzzleException
     */
    public function getUserDepositRecordList(array $data): array
    {
        return $this->client->post('/getUserDepositRecordList', $data);
    }
}

File: ruby/lib/ccpayment/services/user_deposit_service.rb
module CCPayment
  module Services
    class UserDepositService
      def initialize(client)
        @client = client
      end

      def get_or_create_user_deposit_address(data)
        @client.post('/getOrCreateUserDepositAddress', data)
      end

      def get_user_deposit_record(record_id)
        @client.post('/getUserDepositRecord', { recordId: record_id })
      end

      def get_user_deposit_record_list(data)
        @client.post('/getUserDepositRecordList', data)
      end
    end
  end
end
